src/databricks/labs/dqx/check_funcs.py

```diff
@@ -2004,23 +2107,100 @@ def _convert_ipv4_to_bits(ip_col: Column) -> Column:
         prefix_length: Prefix length for CIDR notation
+        total_bits: Total number of bits in the IP address (32 for IPv4)
+
+    Returns:
+        Network address as a binary string of the same total length
+    """
+    return F.rpad(F.substring(ip_bits, 1, prefix_length), total_bits, "0")
+
+
+def _build_is_valid_ipv6_address_udf() -> Callable:
+    """
+    Build a user-defined function (UDF) to check if a string is a valid IPv6 address.
 
     Returns:
-        Network address as a 32-bit binary string
+        Callable: A UDF that checks if a string is a valid IPv6 address
     """
-    return F.rpad(F.substring(ip_bits, 1, prefix_length), 32, "0")
+
+    @F.pandas_udf("boolean")  # type: ignore[call-overload]
+    def _is_valid_ipv6_address_udf(column: pd.Series) -> pd.Series:
+        # Self-contained validation logic to avoid serialization issues
+        def is_valid_ipv6_local(ip_str):
+            if pd.isna(ip_str) or ip_str is None:
+                return False
+            import ipaddress  # pylint: disable=import-outside-toplevel
+
+            try:
+                # Import inside UDF to avoid serialization issues
+                ipaddress.IPv6Address(str(ip_str))
+                return True
+            except (ipaddress.AddressValueError, TypeError):
+                return False
+
+        return column.apply(is_valid_ipv6_local)
+
+    return _is_valid_ipv6_address_udf
+
+
+def _build_is_ipv6_address_in_cidr_udf() -> Callable:
+    """
+    Build a user-defined function (UDF) to check if an IPv6 address is in a CIDR block.
+
+    Returns:
+        Callable: A UDF that checks if an IPv6 address is in a CIDR block
+    """
+
+    @F.pandas_udf("boolean")  # type: ignore[call-overload]
+    def handler(ipv6_column: pd.Series, cidr_column: pd.Series) -> pd.Series:
+        # Self-contained CIDR checking logic to avoid serialization issues
+        def ipv6_in_cidr_local(ip_str, cidr_str):
+            if pd.isna(ip_str) or pd.isna(cidr_str) or ip_str is None or cidr_str is None:
+                return False
+            import ipaddress  # pylint: disable=import-outside-toplevel
+
+            try:
+                # Import inside UDF to avoid serialization issues
+                ip_obj = ipaddress.IPv6Address(str(ip_str))
+                network = ipaddress.IPv6Network(str(cidr_str), strict=False)
+                return ip_obj in network
+            except (ipaddress.AddressValueError, ipaddress.NetmaskValueError, TypeError):
+                return False
+
+        return ipv6_column.combine(cidr_column, ipv6_in_cidr_local)
+
+    return handler
+
+
+def _is_valid_ipv6_cidr_block(cidr: str) -> bool:
+    """Validate if the string is a valid IPv6 CIDR block.
+
+    Args:
+        cidr: The CIDR block string to validate.
+    Returns:
+        True if the string is a valid CIDR block, False otherwise.
+    """
+    import ipaddress  # pylint: disable=import-outside-toplevel
```

Comment:
Had to add disable=import-outside-toplevel if not make lint complains about the import not being on outer level.